src/SegmentedArray.chpl

```diff
@@ -717,6 +717,78 @@ module SegmentedArray {
       return (subbedOffsets, subbedVals, numReplacements);
     }
 
+    /*
+      Strip out all of the leading and trailing characters of each element of a segstring that are
+      called out in the "pattern" argument.
+
+      :arg pattern: pattern used to find matches of the leading and trailing characters in the segstring
+      :type pattern: string
+
+      :arg containsSpace: the string to replace pattern matches with
+      :type replStr: string
+
+      :arg initCount: If count is nonzero, at most count splits occur. If zero, substitute all occurences of pattern
+      :type initCount: int
+
+      :arg containsSpace: If True, appends a space character to the pattern
+      :type containsSpace: bool
+
+      :returns: Strings – substrings with stripped characters from the original string and the offsets into those substrings
+    */
+
+    proc strip(pattern: string, containsSpace: bool) throws {
+      checkCompile(pattern);
+      ref origOffsets = this.offsets.a;
+      ref origVals = this.values.a;
+      const lengths = this.getLengths();
+
+      saLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),
+                                             "pattern: %s - origOffsets: %t - origVals: %t"
+                                             .format(pattern, origOffsets, origVals:bytes));
+
+      var replacedLens: [this.offsets.aD] int;
+      var newPattern: string;
+
+      // Pattern contains only whitespace characters or is empty
+      if containsSpace == true {
+        newPattern = pattern + " ";
+      }
```

Comment:
Again, this is going to be an issue, because we remove all spaces, not just trailing ones. There are a few ways you could get around this with the message parsing, or you could record the position of the spaces, the message parsing will be much easier though.

Reply:
I don't think it should matter because `pattern` is the set of characters being stripped, so position/multiplicity of the characters shouldn't change anything

Reply:
I agree with Pierce that position doesn't matter. Also, I'm not sure what you mean by "remove all spaces" but if there are spaces inside the original string that are bordered by non-matching characters then they are not removed. However, I changed the Python code to remove the "containsSpace" flag.

Reply:
I left this comment by mistake. I had `remove` functionality in my head, not `strip` that is my bad. Sorry!